client/app/operations/deployments/opsDeploymentsList.js

```diff
@@ -7,26 +7,34 @@ angular.module('EnvironmentManager.operations').component('opsDeploymentsList',
   bindings: {
     query: '<',
     showDetails: '&',
-    foundServicesFilter: '&'
+    foundServicesFilter: '&',
   },
   controllerAs: 'vm',
   controller: function ($scope, Deployment, $uibModal) {
     var vm = this;
+    var triggeredRefresh = false;
 
     function refresh() {
       vm.dataLoading = true;
+
       Deployment.getAll(vm.query).then(function (data) {
         vm.deployments = data.map(Deployment.convertToListView);
-
         vm.uniqueServices = _.uniq(data.map(function (d) { return d.Value.ServiceName; }));
-
         vm.dataLoading = false;
         vm.dataFound = true;
+        
+        vm.summary = vm.deployments.reduce(function(summary, d) {
+          summary[d.status]++;
```

Comment:
Could eliminate hard-coded status names by replacing this line with the statement below and initialising `reduce` with empty object.
```
summary = (summary[d.status] || 0) + 1;
```